src/flags.zig

```diff
@@ -296,90 +296,135 @@ fn parse_value(comptime T: type, flag: []const u8, value: [:0]const u8) T {
     };
 
     if (V == []const u8 or V == [:0]const u8) return value;
-    if (V == ByteSize) return parse_value_size(flag, value);
+    if (V == ByteSize) return ByteSize.parse(flag, value);
     if (@typeInfo(V) == .Int) return parse_value_int(V, flag, value);
     if (@typeInfo(V) == .Enum) return parse_value_enum(V, flag, value);
     comptime unreachable;
 }
 
-pub const ByteSize = struct { bytes: u64 };
-fn parse_value_size(flag: []const u8, value: []const u8) ByteSize {
-    assert((flag[0] == '-' and flag[1] == '-') or flag[0] == '<');
+pub const ByteUnit = enum(u64) {
+    bytes = 1,
+    kb = 1024,
+    mb = 1024 * 1024,
+    gb = 1024 * 1024 * 1024,
+    tb = 1024 * 1024 * 1024 * 1024,
+};
 
-    const units = .{
-        .{ &[_][]const u8{ "TiB", "tib", "TB", "tb" }, 1024 * 1024 * 1024 * 1024 },
-        .{ &[_][]const u8{ "GiB", "gib", "GB", "gb" }, 1024 * 1024 * 1024 },
-        .{ &[_][]const u8{ "MiB", "mib", "MB", "mb" }, 1024 * 1024 },
-        .{ &[_][]const u8{ "KiB", "kib", "KB", "kb" }, 1024 },
-    };
+pub const ByteSize = struct {
+    value: u64,
+    unit: ByteUnit,
 
-    const unit: struct { suffix: []const u8, scale: u64 } = unit: inline for (units) |unit| {
-        const suffixes = unit[0];
-        const scale = unit[1];
-        for (suffixes) |suffix| {
-            if (std.mem.endsWith(u8, value, suffix)) {
-                break :unit .{ .suffix = suffix, .scale = scale };
+    pub fn default(value: u64) ByteSize {
+        return .{
+            .value = value,
+            .unit = ByteUnit.bytes,
+        };
+    }
+
+    fn parse(flag: []const u8, value: []const u8) ByteSize {
+        assert((flag[0] == '-' and flag[1] == '-') or flag[0] == '<');
+
+        const units = .{
+            .{ &[_][]const u8{ "TiB", "tib", "TB", "tb" }, ByteUnit.tb },
+            .{ &[_][]const u8{ "GiB", "gib", "GB", "gb" }, ByteUnit.gb },
+            .{ &[_][]const u8{ "MiB", "mib", "MB", "mb" }, ByteUnit.mb },
+            .{ &[_][]const u8{ "KiB", "kib", "KB", "kb" }, ByteUnit.kb },
+        };
+
+        const unit: struct {
+            suffix: []const u8,
+            unit: ByteUnit,
+        } = unit: inline for (units) |unit| {
+            const suffixes = unit[0];
+            const unitkind = unit[1];
+            for (suffixes) |suffix_| {
+                if (std.mem.endsWith(u8, value, suffix_)) {
+                    break :unit .{ .suffix = suffix_, .unit = unitkind };
+                }
             }
-        }
-    } else break :unit .{ .suffix = "", .scale = 1 };
+        } else break :unit .{ .suffix = "", .unit = ByteUnit.bytes };
+
+        assert(std.mem.endsWith(u8, value, unit.suffix));
+        const value_numeric = value[0 .. value.len - unit.suffix.len];
```

Comment:
It can be simplified, returning the input value as string instead of the suffix.

```zig
        const unit: struct {
            value_input: []const u8,
            unit: ByteUnit,
        } = unit: inline for (units) |unit| {
            const suffixes = unit[0];
            const unit_kind = unit[1];
            for (suffixes) |suffix_| {
                if (std.mem.endsWith(u8, value, suffix_)) {
                    break :unit .{ .value_input = value[0 .. value.len - suffix_.len], .unit = unit_kind };
                }
            }
        } else break :unit .{ .value_input = value, .unit = ByteUnit.bytes };
```

Reply:
I have made this change.